### ISSUE TITLE
Remove redundant partial modifier on type declaration

### ISSUE THREAD

**Issue by xtqqczze** [dotnet/roslyn#76784]
**Brief description:**

This analyzer identifies partial types (classes, structs, or interfaces) declared with only a single part in the codebase. The partial modifier is intended for types split across multiple files, and using it unnecessarily can cause confusion or clutter. The analyzer suggests removing the partial modifier unless additional parts are expected.

**Languages applicable:**

This analyzer is applicable to C# and VB.

**Code example that the analyzer should report:**

```csharp
public partial class MyClass
{
    public void DoSomething() { }
}
```

The analyzer should not report when there are multiple parts of the type.

**Additional information:**

https://www.jetbrains.com/help/rider/PartialTypeWithSinglePart.html

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).
